tests/main/snap-confine/task.yaml

```diff
@@ -0,0 +1,15 @@
+summary: Port of the snap-confine integration tests
+
+restore: |
+    rm -f tmpl
+
+execute: |
+    # test_bad_seccomp_filter_args_clone
+    for i in 'CLONE_NEWNE' 'CLONE_NETNETT' 'CL0NE_NEWNET' ; do
+        printf "Test bad seccomp arg filtering (setns - %s)" "$i"
+        echo "setns - $i" >> tmpl
+        if /usr/lib/snapd/snap-seccomp compile tmpl tmpl.bpf; then
+               echo "snap-seccomp should have failed to compile this"
+               exit 1
+        fi
+    done
```

Comment:
This is a pure syntax check and therefore should be covered by the unit tests. All pure syntax checks should be ported to cmd/snap-seccomp/main_test.go while ones testing if confinement is in place should be in spread tests. To make this easier, here are the syntax tests to be ported to main_export.go:
* test_bad_seccomp_filter_*
* test_restrictions_working_args_clone
* test_restrictions_working_args_mknod
* test_restrictions_working_args_prctl
* test_restrictions_working_args_prio
* test_restrictions_working_args_quotactl
* test_restrictions_working_args_socket
* test_restrictions_working_args_termios

and here are the ones to be ported to spread:
* test_complain
* test_complain_missed
* test_unrestricted
* test_unrestricted_missed
* test_noprofile
* test_whitelist (this may already be sufficiently covered by other spread tests)
* test_restrictions (this may already be sufficiently covered by other spread tests)
* test_restrictions_working (this may already be sufficiently covered by other spread tests)
* test_restrictions_working_args

All that said, having one spread test with busted syntax is valuable to make sure things are functionally sane.